### PR TITLE
Add Append Capability to Compressed Level + Test

### PR DESCRIPTION
**Pull request by bharath2438**
@hameerabbasi I've implemented the 4 append functions applicable to compressed level. I wasn't exactly sure of the format of the test case so I've just included all the functions for a base-case, I can add more test cases with other other levels once the format is finalized.

### CODE REVIEW

**Comment by bharath2438** [comment]
@hameerabbasi I've added the test-cases for the four levels which have insert/append.

File: test/source/compressed_test.cpp [hunk 1]
```diff
@@ -110,3 +110,58 @@ TEST_CASE("Compressed-DCSR")
...
+        for (auto const [i2, p2] : s2.iter_helper(std::make_tuple(i1), p1))
+        {
+            CHECK(pos_holder[l1] + l2 == p2);
+            CHECK(crd_holder[p2] == i2);
+            ++l2;
+        }
+        CHECK(l2 == pos_holder[l1 + 1] - pos_holder[l1]);
+        ++l1;
+    }
+    CHECK(l1 == d1.size(ONE));
```
Comment (hameerabbasi) [inline test/source/compressed_test.cpp:166]:
```suggestion
    CHECK(l1 == SIZE1);
```

File: include/xsparse/levels/hashed.hpp [hunk 2]
```diff
@@ -132,19 +132,20 @@ namespace xsparse
...
             {
                 m_crd[pkm1][ik] = pk;
             }
 
-            inline void insert_edges(typename BaseTraits::PKM1 pkm1) const noexcept
+            inline void insert_edges([[maybe_unused]] typename BaseTraits::PKM1 pkm1,
+                                     typename BaseTraits::PK szk) noexcept
             {
-                m_crd_size += m_crd[pkm1].size();
+                m_crd_size += szk;
```
Comment (hameerabbasi) [inline include/xsparse/levels/hashed.hpp:148]: This seems incorrect, but in any case I'd remove this function (and the `sized` property) from `hashed`, it doesn't possess it with our implementation.

File: test/source/dense_test.cpp [hunk 3]
```diff
@@ -49,3 +49,32 @@ TEST_CASE("Dense-2D")
...
+        CHECK(l1 == p1);
+        CHECK(l1 == i1);
+        uintptr_t l2 = 0;
+        for (auto const [i2, p2] : d2.iter_helper(std::make_tuple(i1), p1))
+        {
+            CHECK(l1 * d2.size(ONE) + l2 == p2);
+            CHECK(l2 == i2);
+            ++l2;
+        }
+        CHECK(l2 == d2.size(ONE));
```
Comment (hameerabbasi) [inline test/source/dense_test.cpp:76]:
```suggestion
        CHECK(l2 == SIZE2);
```

File: test/source/dense_test.cpp [hunk 4]
```diff
@@ -49,3 +49,32 @@ TEST_CASE("Dense-2D")
...
+        ++l1;
+    }
+    CHECK(l1 == d1.size(ONE));
```
Comment (hameerabbasi) [inline test/source/dense_test.cpp:79]:
```suggestion
    CHECK(l1 == SIZE1);
```

File: include/xsparse/levels/hashed.hpp (hunk 2 above)
Comment (bharath2438) [reply include/xsparse/levels/hashed.hpp:148]: Okay, so do I remove `insert_edges` and `m_crd_size` from hashed?

Comment (hameerabbasi) [reply include/xsparse/levels/hashed.hpp:148]: Yes.

File: test/source/dense_test.cpp (hunk 4 above)
Comment (hameerabbasi) [reply test/source/dense_test.cpp:79]: That is usually needed only when iterating. You calculate the size of sized levels by passing the amount of inserted coordinates in the last level.

**Comment by hameerabbasi** [comment]
It seems like you mixed up the two branches/PRs. Please separate them if possible by cherry picking commits in git.

**Review by hameerabbasi — APPROVED** [review]

**Comment by hameerabbasi** [comment]
We could improve tests, but this can go in as-is. Merging, thanks, @bharath2438!

**Pull request merged**